### DISCUSSION TITLE
How does mempool keep track of lighting channel funding transactions?

### DISCUSSION THREAD

**Discussion by MikaeelMF** [mempool/mempool#4638] [Q&A] [answered]
I wanted to look into getting information on lighting channel funding transactions. Now as far as I understood by reading lightning node implementations' APIs, you can get info on current observable channels that are visible to your node, but if someone wants to get info on a channel that was closed say 500 blocks ago, and doesn't belong to the current node operator, how should one do that? Because I see that mempool is providing this information through the API "GET Channels from TXID", but if I run my local lightning node, Bitcoin core, and mempool electrs, I won't be able to access that information, right? Is mempool just updating the information as they get announced on lightning network, or there is another source for these closed channels from the past that I'm missing here?

**Comment by softsimon** [comment, marked as answer]
If you run the Mempool backend with Electrs it will take the public graph available from your Lightning node and extract the funding TXID, and periodically scan to see if they are closed, to get the closing status and TXID.

If a channel is already closed and not announced in the network when you start your lightning node the first time, there will be no way of finding this data. Mempool Space has a longer history of storing the lightning network information.

**Reply by MikaeelMF** [reply]
Oh got it! thank you for explanation!
